tests/e2e_tests/test_post_algorithms.py

```diff
@@ -7,117 +7,135 @@
 
 from tests.e2e_tests import algorithms_url
 
-test_cases_post_algorithm_success = [
-    (
-        "logistic_regression",
-        {
-            "inputdata": {
-                "pathology": "dementia",
-                "datasets": ["edsd"],
-                "x": [
-                    "lefthippocampus",
-                    "righthippocampus",
-                    "rightppplanumpolare",
-                    "leftamygdala",
-                    "rightamygdala",
-                ],
-                "y": ["alzheimerbroadcategory"],
-                "filters": {
+from mipengine.controller.api.algorithm_request_dto import (
+    AlgorithmInputDataDTO,
+    AlgorithmRequestDTO,
+)
+
+
+def get_parametrization_list_success_cases():
+
+    parametrization_list = []
+
+    inputdata_dto = AlgorithmInputDataDTO(
+        pathology="dementia",
+        datasets=["edsd"],
+        x=[
+            "lefthippocampus",
+            "righthippocampus",
+            "rightppplanumpolare",
+            "leftamygdala",
+            "rightamygdala",
+        ],
+        y=["alzheimerbroadcategory"],
+        filters={
+            "condition": "AND",
+            "rules": [
+                {
+                    "id": "dataset",
+                    "type": "string",
+                    "value": ["edsd"],
+                    "operator": "in",
+                },
+                {
                     "condition": "AND",
                     "rules": [
                         {
-                            "id": "dataset",
+                            "id": variable,
```

Comment:
???

Reply:
I have not change the value here, if you set the view here in split mode and check the diff closely you ll see the values are as they were, it's just the way git (or github?) shows the diff, that looks like the value of `id` is changed, but it has not. There are 2 `"rules":[{"id":... `in the filters dictionary, one is set to `"dataset"` and one to `variable`
It also caused me some wtfs..